android-json-form-wizard/src/main/java/com/vijay/jsonwizard/activities/JsonFormActivity.java

```diff
@@ -144,6 +144,21 @@ public void onReceive(Context context, Intent intent) {
     private boolean isNextStepRelevant;
     private String nextStep = "";
 
+    private static JSONObject generateTranslatableValue(String value, JSONObject item) throws JSONException {
+        FormUtils formUtils = new FormUtils();
+        JSONObject newValue = new JSONObject();
+        if (item.has(JsonFormConstants.OPTIONS_FIELD_NAME)) {
+            JSONArray options = item.getJSONArray(JsonFormConstants.OPTIONS_FIELD_NAME);
+            JSONObject selectedOption = formUtils.getOptionFromOptionsUsingKey(options, value);
+            newValue.put(JsonFormConstants.VALUE, value);
+            newValue.put(JsonFormConstants.TEXT, selectedOption.optString(JsonFormConstants.TRANSLATION_TEXT, ""));
+            return newValue;
+        }
+        newValue.put(JsonFormConstants.VALUE, value);
+        newValue.put(JsonFormConstants.TEXT, item.optString(JsonFormConstants.TRANSLATION_TEXT, ""));
+        return newValue;
+    }
+
```

Comment:
Isn't this method already in the Utils? Please reuse that

Reply:
If not please move it to the Utils

Reply:
Also add Java Docs for any public methods you write

Reply:
Hi, this was moved to utils in the new widgets touched. The function is already in Utils